### PR TITLE
fix: learning curve tuning [DET-4375, DET-4692, DET-4711]

### PR DESCRIPTION
**Pull request by hkang1**
## Description

Tuning Learning Curve

- Thicken the learning curve. Previously the lines are a bit too thin.
- Add glasbey color palette for the learning curve series.
- Add glasbey color palette for the learning curve trial hp table.
- Place the spinner lower when there is a notice
- Make metric of interest sticky
- Add better messaging for not supporting visualizations for single trial experiments

Fixes

- Check to make sure new data points do not reset the zoom
- Fix long metric axis labels getting cut off
- When you hover out of a highlighted learning curve series, it keeps the corresponding trial table row highlighted. Update to correct this bug.
- Move crosshair for the learning curve chart below the info tooltip
 
Screen Shot 2021-01-05 at 4 53 15 PM

## Checklist

- [ ] User-facing API changes need the "User-facing API Change" label.
- [ ] Release notes should be added as a separate file under `docs/release-notes/`.
See [Release Note](../docs/release-notes/README.md) for details.

[DET-1234]: https://determinedai.atlassian.net/browse/DET-1234

### CODE REVIEW

File: webui/react/src/pages/ExperimentDetails/ExperimentVisualization/LearningCurve.tsx [hunk 1]
```diff
@@ -213,23 +226,39 @@ const LearningCurve: React.FC<Props> = ({
...
   }, [ experiment.id, maxTrials, selectedMetric ]);
 
-  if (pageError) {
+  if (pageError?.message.includes('single-trial experiments are not supported')) {
+    return <Alert
+      description={<>
+        Please consider using multi-trial searchers such as &nbsp;
+        <Link
+          path="/docs/reference/experiment-config.html#adaptive-asha"
+          popout
```
Comment (hamidzr) [inline webui/react/src/pages/ExperimentDetails/ExperimentVisualization/LearningCurve.tsx:235]: blocking: these need `external` tags otherwise won't work

**Review by hamidzr — CHANGES_REQUESTED** [review]
LGTM. 

highlighting some trial isn't that distinctive. I scrolled over a row and was expecting to find the trial highlighted but it wasn't easy to tell. with some colors, it works great. here one is highlighted:
Screen Capture_select-area_20210108164949



Screen Capture_select-area_20210108165456

suggestion: Hyperparameter visua..
@ahj-determined any wording suggestions here? or maybe we link to the searcher section http://localhost:8080/docs/reference/experiment-config.html#searcher 

this looks like a good place to advertise for hp search

**Comment by hamidzr** [comment]
added comment about whether we want to add the new color scheme to overview and maybe the existing experiment page.

**Comment by hamidzr** [comment]
Screen Capture_select-area_20210108170416

**Comment by hamidzr** [comment]
a bunch of nits

- highlighting (from plot to row) only happens on dots
- jagged line, blurry. can we up the resolution? is uplot canvas based? 
250%
Screen Capture_select-area_20210108170918
  - maybe an option would be move the axis labels out of the chart later if we want it to match the rest of the page easier.
100%
Screen Capture_select-area_20210108170653


- zoom between the data points and you don't see any lines. this is understandable but could it be confusing?

**Comment by hamidzr** [comment]
nit: chart and the trial hyperparameters table rerender visibly on changing the metric or number of top trials

**Comment by ahj-determined** [comment]
> @ahj-determined any wording suggestions here? or maybe we link to the searcher section http://localhost:8080/docs/reference/experiment-config.html#searcher

Yeah, I like those two ideas. How about something like:

Hyperparameter visualizations are not applicable for single trial experiments.
Learn about [how to run a hyperparameter search](https://docs.determined.ai/latest/how-to/hyperparameter-tuning.html?highlight=hyperparameter%20search).

File: webui/react/src/pages/ExperimentDetails/ExperimentVisualization/LearningCurve.tsx (hunk 1 above)
Comment (hkang1) [reply webui/react/src/pages/ExperimentDetails/ExperimentVisualization/LearningCurve.tsx:235]: good catch, thanks!

**Comment by hkang1** [comment]
> Overall it looks good to me. there is a small blocker.
> 
> Also:
> 
> nit: highlighting some trial isn't that distinctive. I scrolled over a row and was expecting to find the trial highlighted but it wasn't easy to tell and that could confuse users?. with some colors, it works great. here one is highlighted:
> Screen Capture_select-area_20210108164949
> 
I've reduced the unfocused alpha by a little bit to make the focused on stand out more.

> Screen Capture_select-area_20210108165456
> 
> suggestion: Hyperparameter visua..
> @ahj-determined any wording suggestions here? or maybe we link to the searcher section http://localhost:8080/docs/reference/experiment-config.html#searcher
> 
> this looks like a good place to advertise for hp search

I've updated to the copy @ahj-determined provided, thanks!

**Comment by hkang1** [comment]
> not directly related:
> Screen Capture_select-area_20210108170416

oops, fixed it this time. (incorrect scss module scope)

**Comment by hkang1** [comment]
> a bunch of nits
> 
> * highlighting (from plot to row) only happens on dots
>
The series lines unfortunately are not interactive. I think this is reasonable considering this is a canvas based chart and building in line proximity detection can be expensive for a performance based chart that can have tens of thousands of lines.

> * jagged line, blurry. can we up the resolution? is uplot canvas based?
>   250%
>   Screen Capture_select-area_20210108170918
>   
I couldn't find an option for uplot for this. There is a devicePixelRatio for HTML canvases but that's not a knob for us to change so far as I know... Let me know if you find some options regarding this.d

>   * maybe an option would be move the axis labels out of the chart later if we want it to match the rest of the page easier.
>     100%
>     Screen Capture_select-area_20210108170653
>
This is certainly an option but I don't think it gets us much considering the overhead of managing the labels ourselves. Having uPlot manage allows it to dynamically figured out the layout in relation to the chart and axes labels. For styling the labels to match the page, we can actually use fonts, but I haven't looked too much into this yet...

> * zoom between the data points and you don't see any lines. this is understandable but could it be confusing?
>
I don't think it's supposed to lose the lines when you zoom in. Can't seem to replicate this behavior, maybe I am misunderstanding.

**Review by hamidzr — APPROVED** [review]
LGTM. minus the waiting messaging on terminal states https://github.com/determined-ai/determined/pull/1753#issuecomment-758103912

thanks @hkang1

**Comment by hkang1** [comment]
> LGTM. minus the waiting messaging on terminal states [#1753 (comment)](https://github.com/determined-ai/determined/pull/1753#issuecomment-758103912)
> 
> thanks @hkang1

terminal experiment messaging added, thanks for the quick review @hamidzr !

**Pull request merged**